File: Fitch/Texture2D.cs
namespace Fitch
{
    public struct Texture2D
    {

        private int id;
        private int width;
        private int height;

        public int ID { get { return id; }}
        public int Width { get { return width; } set { width = value; } }
        public int Height { get { return height; } set { height = value; } }

        public Texture2D(int id, int width, int height)
        {

            this.id = id;
            this.width = width;
            this.height = height;

        }
    }
}

File: graphics/Texture2D.cpp
//
// Created by jacom on 2018/08/22.
//

#include "Texture2D.h"

Texture2D::Texture2D() : ID(0), width(0), height(0) {}
Texture2D::Texture2D(GLuint ID, int width, int height) : ID(ID), width(width), height(height) {}